### PR TITLE
Fix error handling and add alert messages

### LINKED ISSUE #28: Error Handling for fetches

### PR DESCRIPTION
**Pull request by ramyabuva**
Creates alerts for errors, or displays the appropriate message to the user. 
https://screenshot.googleplex.com/vmchguqH1gt.png
Closes #28

### CODE REVIEW

File: src/main/webapp/js/courseSelector.js [hunk 1]
```diff
@@ -23,21 +23,25 @@ export const CollegePlanner = (() => {
...
-    option.selected = true;
-    option.hidden = true;
-    departmentContainer.appendChild(option);
-    // Add each course to course list
-    const departmentsDetailed = departmentList.departments;
-    departmentsDetailed.forEach(
-        department => addDepartmentOption(department, departmentContainer));
+    try {
+      const response = await fetch('/api/departments');
+      const departmentList = await response.json();
```
Comment (savsa) [inline src/main/webapp/js/courseSelector.js]: Just put the `await` lines inside the try catch? The other lines below these won't cause an error. And having everything inside the try catch causes another level of indentation

File: src/main/webapp/js/courseSelector.js [hunk 2]
```diff
@@ -61,21 +65,26 @@ export const CollegePlanner = (() => {
...
-    // Add default option to courses dropdown
-    const option = document.createElement('option');
-    option.innerText = 'Select a Course';
-    option.selected = true;
-    option.hidden = true;
-    courseContainer.appendChild(option);
-    // Add each course to course list
-    const coursesDetailed = courseList.courses;
-    coursesDetailed.forEach(course => addCourseOption(course, courseContainer));
+    try {
```
Comment (savsa) [inline src/main/webapp/js/courseSelector.js:76]: same here

File: src/main/webapp/js/planner.js [hunk 3]
```diff
@@ -26,14 +26,44 @@ async function getPlan() {
     selectedClasses: selectedClasses,
     semesters: document.getElementById('semesters').value
   };
-  const response = await fetch('/api/planner', {
-    method: 'POST',
-    body: JSON.stringify(data),
-  });
+  courseContainer.innerText = '';
+  try {
```
Comment (savsa) [inline src/main/webapp/js/planner.js:32]: same here

**Review by savsa — APPROVED** [review]

File: src/main/webapp/js/planner.js [hunk 4]
```diff
@@ -26,14 +26,46 @@ async function getPlan() {
...
+    response = await fetch('/api/planner', {
+      method: 'POST',
+      body: JSON.stringify(data),
+    });
+    courseList = await response.json();
+  } catch (err) {
+    alert('An error occurred.');
+    return;
+  }
+  if (response.ok) {
```
Comment (wli) [inline src/main/webapp/js/planner.js:42]: isn't this implicitly ok because no error was thrown?

File: src/main/webapp/js/planner.js [hunk 5]
```diff
@@ -26,14 +26,46 @@ async function getPlan() {
     selectedClasses: selectedClasses,
     semesters: document.getElementById('semesters').value
   };
-  const response = await fetch('/api/planner', {
-    method: 'POST',
-    body: JSON.stringify(data),
-  });
+  courseContainer.innerText = '';
```
Comment (wli) [inline src/main/webapp/js/planner.js]: write a comment explaining why this line is here (clear error message?)

File: src/main/webapp/js/courseSelector.js [hunk 6]
```diff
@@ -64,9 +71,16 @@ export const CollegePlanner = (() => {
...
-        `/api/courses?department=${encodeURIComponent(selectedDepartment)}`);
-    const courseList = await response.json();
+    let response;
+    let courseList;
+    try {
+      response = await fetch(
+          `/api/courses?department=${encodeURIComponent(selectedDepartment)}`);
+      courseList = await response.json();
+    } catch (err) {
+      alert('An error occurred.');
```
Comment (wli) [inline src/main/webapp/js/courseSelector.js]: is there a reason we can't `createError()` for an HTML thing here too?

**Review by wli — CHANGES_REQUESTED** [review]

File: src/main/webapp/js/planner.js (hunk 4 above)
Comment (ramyabuva) [reply src/main/webapp/js/planner.js:42]: the errors are only thrown if there's an error when getting the response, so this catches any responses with a status that is not in the 200s.

Comment (wli) [reply src/main/webapp/js/planner.js:42]: ah ok, sounds good, thanks for the clarification.

File: src/main/webapp/js/planner.js [hunk 7]
```diff
@@ -26,14 +26,31 @@ async function getPlan() {
...
+  let response;
+  let courseList;
+  try {
+    response = await fetch('/api/planner', {
+      method: 'POST',
+      body: JSON.stringify(data),
+    });
+    courseList = await response.json();
+  } catch (err) {
+    createAlert('An error occurred', 'danger', courseContainer);
```
Comment (wli) [inline src/main/webapp/js/planner.js]: is this supposed to be `CollegePlanner.createAlert`?

File: src/main/webapp/js/courseSelector.js [hunk 8]
```diff
@@ -140,6 +154,22 @@ export const CollegePlanner = (() => {
...
   }
+
+/**
+ * Creates an alert with the specified message in the container
+ * @param {string} message The message string you want to be displayed
+ * @param {string} type type of alert you want to display (primary, secondary,
+ *     success, warning, danger)
+ * @param {Object} container the container you want to display the alert in
+ */
+function createAlert(message, type, container) {
```
Comment (wli) [inline src/main/webapp/js/courseSelector.js:165]: In theory, we should probably put this function in another file entirely (we should have access to it even if courseSelector isn't here), but this is fine for now. We can move it later if needed.

File: src/main/webapp/js/courseSelector.js [hunk 9]
```diff
@@ -140,6 +154,22 @@ export const CollegePlanner = (() => {
     liElement.appendChild(deleteButtonElement);
     return liElement;
   }
+
+/**
+ * Creates an alert with the specified message in the container
+ * @param {string} message The message string you want to be displayed
+ * @param {string} type type of alert you want to display (primary, secondary,
+ *     success, warning, danger)
+ * @param {Object} container the container you want to display the alert in
```
Comment (wli) [inline src/main/webapp/js/courseSelector.js]: `Object` -> `Element`

**Review by wli — APPROVED** [review]

**Comment by wli** [comment]
@googlebot double check cla

**Review by triplehair — APPROVED** [review]

**Pull request merged**